src/library/tabledelegates/stardelegate.cpp

```diff
@@ -1,16 +1,32 @@
 #include "library/tabledelegates/stardelegate.h"
 
 #include <QTableView>
+#include <QTimer>
 
 #include "library/starrating.h"
 #include "library/tabledelegates/stareditor.h"
 #include "library/tabledelegates/tableitemdelegate.h"
 #include "moc_stardelegate.cpp"
+#include "widget/wtracktableview.h"
 
 StarDelegate::StarDelegate(QTableView* pTableView)
         : TableItemDelegate(pTableView),
-          m_isOneCellInEditMode(false) {
+          m_persistentEditorState(PersistentEditor_NotOpen) {
+    connect(this, &QAbstractItemDelegate::closeEditor, this, &StarDelegate::closingEditor);
     connect(pTableView, &QTableView::entered, this, &StarDelegate::cellEntered);
+    connect(pTableView, &QTableView::viewportEntered, this, &StarDelegate::cursorNotOverAnyCell);
+
+    auto* pTrackTableView = qobject_cast<WTrackTableView*>(pTableView);
+    if (pTrackTableView) {
```

Comment:
Is this supposed to always used in a `WTrackTableView`? If yes, it might be used to use a `VERIFY_OR_DEBUG_ASSERT` instead of the condition so we can detect misusage of this delegated in the future

Reply:
I'll do you one better: `StarDelegate::StarDelegate()` should be updated to simply take `WTrackTableView` in the first place - as its caller already does the mentioned `VERIFY_OR_DEBUG_ASSERT`: https://github.com/mixxxdj/mixxx/blob/68d3796d1d776bace4489363a13b4b63b2204891/src/library/basetracktablemodel.cpp#L478-L481